lib/sidekiq/worker_killer.rb

```diff
@@ -89,38 +89,49 @@ def skip_shutdown?(worker, job, queue)
     @skip_shutdown.respond_to?(:call) && @skip_shutdown.call(worker, job, queue)
   end
 
-  def request_shutdown
+  def request_shutdown(worker, job, queue)
     # In another thread to allow underlying job to finish
     Thread.new do
       # Only if another thread is not already
       # shutting down the Sidekiq process
-      shutdown if MUTEX.try_lock
+      shutdown(worker, job, queue) if MUTEX.try_lock
     end
   end
 
-  def shutdown
+  def shutdown(worker, job, _queue)
     warn "sending quiet to #{identity}"
     sidekiq_process.quiet!
```

Comment:
We discussed this before we left yesterday but if we quiet we'll need to terminate the process as it will stop picking up new work. I couldn't find a way to reverse the quiet so the restart will be the only way.

Reply:
I just added the `stop!` back. Is that enough, or do I have to add `::Process.kill` back?

Reply:
Stop should be enough as it will kill after it's own shutdown timeout.